main.go

```diff
@@ -219,22 +219,52 @@ func getBestThumbnail(path string) (newpath string) {
 	return strings.Replace(path, "maxres.jpg", "mqdefault.jpg", 1)
 }
 
+// exists returns whether the given file or directory exists
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
 func main() {
-	socket := "socket" + string(os.PathSeparator) + "http-proxy.sock"
-	syscall.Unlink(socket)
-	listener, err := net.Listen("unix", socket)
+	env_port := os.Getenv("PORT")
+	port := "8080"
+
+	if env_port != "" {
+		port = env_port
+	}
+
 	srv := &http.Server{
 		ReadTimeout:  5 * time.Second,
 		WriteTimeout: 10 * time.Second,
-		Addr:         ":8080",
+		Addr:         ":" + port,
 		Handler:      &requesthandler{},
 	}
+
+	exists, err := exists("socket/")
+
 	if err != nil {
-		fmt.Println("Failed to bind to UDS, falling back to TCP/IP")
-		fmt.Println(err.Error())
-		srv.ListenAndServe()
-	} else {
+		log.Panicln(err)
+	}
+
+	if env_port == "" && exists {
+		socket := "socket" + string(os.PathSeparator) + "http-proxy.sock"
+		syscall.Unlink(socket)
+		listener, err := net.Listen("unix", socket)
+
+		if err != nil {
+			log.Panicln(err)
+		}
+
 		defer listener.Close()
 		srv.Serve(listener)
+	} else {
+		fmt.Println("WARNING: You are listening on a HTTP Socket, you are strongly recommended to use a Unix Domain Socket for performance reasons.")
```

Comment:
I wouldn't say it's quite necessary to be this strong. Especially given that in many use cases, using unix sockets is impossible. To a novice user, this could be quite harmful.

Reply:
`Notice` or `Note` would be more appropriate than a warning.

Reply:
In the official guides, https://piped-docs.kavin.rocks/docs/self-hosting/#docker-compose-caddy-aio-script and https://piped-docs.kavin.rocks/docs/self-hosting/#docker-compose-with-nginx, it's always been recommended to use a Unix socket, so I don't see why this is unreasonable. Never should you run a public instance this way.

Reply:
```suggestion
		fmt.Println("Note: You are listening on a HTTP Socket. Listening on a Unix Domain Socket may increase performance.")
```

Reply:
> Never should you run a public instance this way.

There's a difference between running with a port, and exposing that port to the internet directly without a reverse proxy infront of it.

Reply:
> There's a difference between running with a port, and exposing that port to the internet directly without a reverse proxy infront of it.

That's not the point here, even when using TCP/IP within localhost / between the reverse proxy and ytproxy, you're still wasting a significant amount of CPU resources unnecessarily.

Reply:
@FireMasterK In deployment scenarios with a service mesh (a la Istio or Consul Connect) or if one has particular reasons to keep the two processes separated, it can be perfectly fine to prefer HTTP sockets.

Agreed that this is too strong and risks hurting more than it helps (unless there is some particular nuance specific to ytproxy I am not aware of)